contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -393,23 +383,38 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
      */
     function sendSpecifiedReceipts(
         bytes32 originBlockchainID,
-        uint256[] calldata messageIDs,
+        bytes32[] calldata messageIDs,
         TeleporterFeeInfo calldata feeInfo,
         address[] calldata allowedRelayerAddresses
-    ) external senderNonReentrant returns (uint256) {
+    ) external senderNonReentrant returns (bytes32) {
+        // If the blockchain ID has yet to be initialized, no messages have ever been received by
+        // this contract, meaning that the receipt(s) will not be found in any event.
+        // Thus, don't need to initialize the blockchain ID here.
+        bytes32 blockchainID_ = blockchainID;
```

Comment:
Even so, I'm not a fan of selective treatment of a lazily initialized variable. Ideally, initialization would be behind an interface that completely abstracts the underlying variable state, and that variable would not be able to be directly accessed. We could probably implement something like that here, though that's probably more heavyweight than its worth. In the absence of enforcing that at the compiler level, I think we should fetch `blockchainID_` in a consistent way _everywhere_ it's used, even if we can guarantee it's initialized. That makes the contract much easier to reason about, even at a marginally higher gas cost for some methods. My 2c.

Reply:
I don't feel strongly either way here. 

The one observation I'd call out is that it's impossible for the blockchain ID to be initialize as a part of `sendSpecifiedReceipts`, `retryMessageExecution`, or `retrySendCrossChainMessage`, because if it has not previously been initialized, any calls to those functions will revert in any event, leaving the value uninitialized afterwards even if `initializeBlockchainID` is called.

I agree having a uniform access pattern everyone is less complicated and still correct OTOH though. Ultimately doesn't make a big difference to me as long as the comments make it clear.

Reply:
I agree in general that we should avoid code-level optimizations for small gas savings if they hurt readability. It really should be the job of the compiler to perform these optimizations. With that said, because of the reverts that Michael pointed out, the functions will need comments about initialized/uninitialized variables no matter which way we go. So perhaps just having the comments is enough.

Reply:
> it's impossible for the blockchain ID to be initialize as a part of sendSpecifiedReceipts, retryMessageExecution, or retrySendCrossChainMessage

That's a good point. I'm ok with leaving as is and denoting this behavior with comments.